src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -429,25 +441,20 @@ private function registerWorkflowConfiguration(array $workflows, ContainerBuilde
                 $markingStoreDefinition = new Reference($workflow['marking_store']['service']);
             }
 
-            $definitionDefinition->addTag('workflow.definition', array(
-                'name' => $name,
-                'type' => $type,
-                'marking_store' => isset($workflow['marking_store']['type']) ? $workflow['marking_store']['type'] : null,
-            ));
-            $definitionDefinition->setPublic(false);
-
+            // Create Workflow
             $workflowDefinition = new DefinitionDecorator(sprintf('%s.abstract', $type));
             $workflowDefinition->replaceArgument(0, $definitionDefinition);
             if (isset($markingStoreDefinition)) {
                 $workflowDefinition->replaceArgument(1, $markingStoreDefinition);
             }
             $workflowDefinition->replaceArgument(3, $name);
 
+            // Store to container
             $workflowId = sprintf('%s.%s', $type, $name);
-
-            $container->setDefinition(sprintf('%s.definition', $workflowId), $definitionDefinition);
             $container->setDefinition($workflowId, $workflowDefinition);
+            $container->setDefinition(sprintf('%s.definition', $workflowId), $definitionDefinition);
```

Comment:
We need this line or else we can grab it from the container in the compiler pass.